src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/PackageRestoreTelemetryService.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.ComponentModel.Composition;
+
+namespace Microsoft.VisualStudio.Telemetry
+{
+    [Export(typeof(IPackageRestoreTelemetryService))]
+    internal class PackageRestoreTelemetryService : IPackageRestoreTelemetryService
+    {
+        private readonly ITelemetryService _telemetryService;
+
+        [ImportingConstructor]
+        public PackageRestoreTelemetryService(ITelemetryService telemetryService)
+        {
+            _telemetryService = telemetryService;
+        }
+
+        private string GetProjectId(string fullPath) => string.IsNullOrEmpty(fullPath) ? string.Empty : _telemetryService.HashValue(fullPath);
+
+        /// <summary>
+        /// Posts a telemetry events from package restore components.
+        /// </summary>
+        /// <param name="packageRestoreOperationName">The name of the specific package restore operation.</param>
+        /// <param name="fullPath">The full path to the project that needs a package restore.</param>
+        public void PostPackageRestoreEvent(string packageRestoreOperationName, string fullPath)
```

Comment:
Nit: `fullPath` might be better named as `projectFullPath` since all the calls use a project's path. Also, the description of the parameter says it is a project path too.

Reply:
If the `UnconfiguredProject` was imported into the constructor, I think these `fullPath` parameters could be removed.